pkg/types/chain_reader.go

```diff
@@ -0,0 +1,56 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc/status"
+)
+
+type errChainReader string
+
+func (e errChainReader) Error() string { return string(e) }
+
+const (
+	InvalidTypeError   = errChainReader("invalid type")
+	FieldNotFoundError = errChainReader("field not found")
+	InvalidConfigError = errChainReader("invalid configuration")
+	UnsupportedError   = errChainReader("unsupported")
```

Comment:
These are sentinel errors, which the google style guide recommend be named with an `Err` prefix:
```suggestion
	ErrInvalidType   = errChainReader("invalid type")
	ErrFieldNotFound = errChainReader("field not found")
	ErrInvalidConfig = errChainReader("invalid configuration")
	ErrUnsupported   = errChainReader("unsupported")
```

Reply:
Why do we need to define our own `UnsupportedError`? The standard library provides `errors.ErrUnsupported`, and grpc has its own standards as well. Is there any way to leverage those existing standards?

Reply:
Just seemed simpler to have all the errors chainlink can return handled in the same way, rather than adding a special case.

I see there's an Unimplemented gRPC status code.  Changing it to look for the during the unwrap and convert it into `errors.ErrUnsupported`